---
title: "ER substrate specificity analysis"
output: html_notebook
---

### Analysis for the Amniote Orthologs

```{r}
Prefix <- "~/projects/ER_proj/"
```

```{r}
#####EM genes with complementary activity


######Histone Methylation
H3K4_writers_hm <- c("KMT2A", "KMT2C", "KMT2D", "SETD1A", "SETD1B", 
                     "SETD7", "SMYD1", "SMYD2", "ASH1L", "PRDM9")#"KMT2B",
H3K27_writers_hm <- c("EZH1", "EZH2")
H3K36_writers_hm <- c("NSD1", "WHSC1", "WHSC1L1", "SETD2", "SMYD2", "ASH1L", "SETD3", "SETMAR")
H3K9_writers_hm <- c("PRDM2", "EHMT1", "EHMT2", "SETDB1", "SUV39H1")

H3K4_erasers_hm <- c("KDM1A", "KDM1B", "KDM5A", "KDM5B", "KDM5C", "KDM5D")#, "NO66"
H3K27_erasers_hm <- c("KDM6A", "UTY", "KDM6B", "KDM7A", "PHF8")
H3K36_erasers_hm <- c("KDM2A", "KDM2B", "KDM4A", "KDM4B", "KDM4C", "KDM4D")
H3K9_erasers_hm <- c("KDM3A", "KDM3B", "JMJD1C", "KDM4A", "KDM4B", "KDM4C", "KDM4D", "PHF8", "PHF2")


######Histone Acetylation
H3K27_writers_hac <- c("EP300", "CREBBP")

H3K9_erasers_hac <- c("SIRT1", "SIRT2")


########
EM_specificities <- data.frame(Symbol = c(H3K4_writers_hm, H3K27_writers_hm, 
                                             H3K36_writers_hm, H3K9_writers_hm, H3K4_erasers_hm, 
                                             H3K27_erasers_hm, H3K36_erasers_hm, H3K9_erasers_hm, 
                                             H3K27_writers_hac, H3K9_erasers_hac), 
                               Category = c(rep("H3K4_methylation_writer", length(H3K4_writers_hm)), 
                                               rep("H3K27_methylation_writer", length(H3K27_writers_hm)), 
                                               rep("H3K36_methylation_writer", length(H3K36_writers_hm)), 
                                               rep("H3K9_methylation_writer", length(H3K9_writers_hm)), 
                                               rep("H3K4_methylation_eraser", length(H3K4_erasers_hm)), 
                                               rep("H3K27_methylation_eraser", length(H3K27_erasers_hm)), 
                                               rep("H3K36_methylation_eraser", length(H3K36_erasers_hm)), 
                                               rep("H3K9_methylation_eraser", length(H3K9_erasers_hm)), 
                                               rep("H3K27_acetylation_writer", length(H3K27_writers_hac)), 
                                               rep("H3K9_acetylation_eraser", length(H3K9_erasers_hac))))

```



### EXTRACTING SPECIFIC GENES FROM GTEx TPM File

1. Arguments introduced for the execution
```{r}
Prefix <- "~/projects/ER_proj/"
setwd(Prefix)
args <- commandArgs(trailingOnly = TRUE) #Argument for first input
input1 <- "GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_median_tpm.gct"
input2 <- "human_id_symbol_class.tsv"
```

2. Importing and manipulating the data
```{r}
#Load the library
library(dplyr)


#File name I will use GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_median_tpm.gct
data<-read.delim(paste0(Prefix,"part2/", input1)) #Import data
colnames(data)<- gsub("\\.","",colnames(data))#Remove ... and . from colnames introduced by R
data$gene_id<-gsub("\\..*","",data$gene_id) #Remove .* from ENSEMBL IDs

#We need to make following renamings for ease in data analysis
#R script for replacements
colnames(data)<- gsub("AdiposeSubcutaneous","Adipose_SUB",colnames(data))
colnames(data)<- gsub("AdiposeVisceralOmentum","Adipose_VOM",colnames(data))
colnames(data)<- gsub("AdrenalGland","Adrenal",colnames(data))
colnames(data)<- gsub("ArteryAorta","Artery_AOR",colnames(data))
colnames(data)<- gsub("ArteryCoronary","Artery_COR",colnames(data))
colnames(data)<- gsub("ArteryTibial","Artery_TIB",colnames(data))
colnames(data)<- gsub("BrainAmygdala","Brain_AMY",colnames(data))
colnames(data)<- gsub("BrainAnteriorcingulatecortexBA24","Brain_ACC",colnames(data))
colnames(data)<- gsub("BrainCaudatebasalganglia","Brain_CBG",colnames(data))
colnames(data)<- gsub("BrainCerebellarHemisphere","Cerebellum_HEM",colnames(data))
colnames(data)<- gsub("BrainCerebellum","Cerebellum_CER",colnames(data))
colnames(data)<- gsub("BrainCortex","Brain_COR",colnames(data))
colnames(data)<- gsub("BrainFrontalCortexBA9","Brain_FRO",colnames(data))
colnames(data)<- gsub("BrainHippocampus","Brain_HIP",colnames(data))
colnames(data)<- gsub("BrainHypothalamus","Brain_HYP",colnames(data))
colnames(data)<- gsub("BrainNucleusaccumbensbasalganglia","Brain_NBG",colnames(data))
colnames(data)<- gsub("BrainPutamenbasalganglia","Brain_PBG",colnames(data))
colnames(data)<- gsub("BrainSpinalcordcervicalc1","Brain_SPI",colnames(data))
colnames(data)<- gsub("BrainSubstantianigra","Brain_SUB",colnames(data))
colnames(data)<- gsub("BreastMammaryTissue","Mammary",colnames(data))
colnames(data)<- gsub("CellsEBVtransformedlymphocytes","Cells_EBV",colnames(data))
colnames(data)<- gsub("CellsTransformedfibroblasts","Cells_FIB",colnames(data))
colnames(data)<- gsub("CervixEctocervix","Cervix_ECT",colnames(data))
colnames(data)<- gsub("CervixEndocervix","Cervix_END",colnames(data))
colnames(data)<- gsub("ColonSigmoid","Colon_SIG",colnames(data))
colnames(data)<- gsub("ColonTransverse","Colon_TRA",colnames(data))
colnames(data)<- gsub("EsophagusGastroesophagealJunction","Esophagus_GAS",colnames(data))
colnames(data)<- gsub("EsophagusMucosa","Esophagus_MUC",colnames(data))
colnames(data)<- gsub("EsophagusMuscularis","Esophagus_MUS",colnames(data))
colnames(data)<- gsub("HeartAtrialAppendage","Heart_ATR",colnames(data))
colnames(data)<- gsub("HeartLeftVentricle","Heart_LVE",colnames(data))
colnames(data)<- gsub("SkinNotSunExposedSuprapubic","Skin_NON",colnames(data))
colnames(data)<- gsub("SkinSunExposedLowerleg","Skin_SUN",colnames(data))
colnames(data)<- gsub("SmallIntestineTerminalIleum","Small_Intestine",colnames(data))
colnames(data)<- gsub("KidneyCortex","Kidney",colnames(data))
colnames(data)<- gsub("MuscleSkeletal","Skeletal_Muscle",colnames(data))


data2<-data[,3:dim(data)[2]]#Cut the matrix before ordering it by columns (to avoid gene_id and Description columns)
data3<-data2[ , order(names(data2))] #Order the columns by column names
data4<-cbind(data[1],data[2],data3) #Add first two columns back to the matrix

```

3. Importing the gene list and joining them with the initial input
```{r}
ensembl <- read.table(paste0(Prefix,"part2/", input2), sep="\t",header=TRUE)#import the ensembl file that contains ENSEMBL ID and matching GeneNames
EM_specificities <- join(EM_specificities, ensembl, by = "Symbol")
```


```{r}
joined<- plyr::join(data4, EM_specificities, by="gene_id") #use join function to add a column of gene names/Class corresponding to the ENSEMBL ID to the last column
joined2<- na.omit(joined, cols="Category") #Remove the rows that contain non-matching genes
#joined2 is tha expression file for ER Machinery Proteins (RMMs)
replaced <- joined2 %>% dplyr::select(Category, everything()) #place the last column to first column
replaced2 <- replaced %>% dplyr::select(Symbol, everything())#place the last column to first column
replaced2$Description <- NULL#Remove the redundant column (Gene names)
replaced3<-replaced2[order(replaced2$Category),] #Sort by Class


#Removing the Cell data because they introduce bias to tissue-wide expression analysis
replaced3[,"Cells_EBV"]<-NULL #Remove Cells 
replaced3[,"Cells_FIB"]<-NULL #Remove Cells
replaced3[,"WholeBlood"]<-NULL #Remove Whole Blood
write.table(replaced3,file="ER.GTEX.TPM.tsv",quote=FALSE, sep="\t",row.names=FALSE) #Export TPM for specific genes

#Take mean of tissues with multiple parts (Brain, Cerebellum etc.)
replaced3$Brain <- rowMeans(replaced3[,grep("^Brain_",colnames(replaced3))],na.rm = TRUE) #Mean of the columns that have matching name
replaced3[,grep("^Brain_",colnames(replaced3))]<- NULL #remove those columns after adding mean column
replaced3$Cerebellum <- rowMeans(replaced3[,grep("^Cerebellum_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Cerebellum_",colnames(replaced3))]<- NULL
replaced3$Adipose <- rowMeans(replaced3[,grep("^Adipose_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Adipose_",colnames(replaced3))]<- NULL
replaced3$Artery <- rowMeans(replaced3[,grep("^Artery_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Artery_",colnames(replaced3))]<- NULL
replaced3$Cervix <- rowMeans(replaced3[,grep("^Cervix_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Cervix_",colnames(replaced3))]<- NULL
replaced3$Colon <- rowMeans(replaced3[,grep("^Colon_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Colon_",colnames(replaced3))]<- NULL
replaced3$Esophagus <- rowMeans(replaced3[,grep("^Esophagus_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Esophagus_",colnames(replaced3))]<- NULL
replaced3$Heart <- rowMeans(replaced3[,grep("^Heart_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Heart_",colnames(replaced3))]<- NULL
replaced3$Skin <- rowMeans(replaced3[,grep("^Skin_",colnames(replaced3))],na.rm = TRUE)
replaced3[,grep("^Skin_",colnames(replaced3))]<- NULL

#Order the columns by alphabetical order again
replaced4<-replaced3[,4:dim(replaced3)[2]]#Cut the matrix before ordering it by columns (to avoid gene_id and Description columns)
replaced5<-replaced4[ , order(names(replaced4))] #Order the columns by column names
final<-cbind(replaced3[1],replaced3[2],replaced3[3], replaced5) #Add first two columns back to the matrix
write.table(final,file="ER.GTEX.TissueAveraged.TPM.tsv",quote=FALSE, sep="\t",row.names=FALSE) #Export TPM for specific genes in averaged tissues

```



### Tissue-wide expression plots using GTEx values

1. Arguments introduced for the execution
```{r}
input1 <- "ER.GTEX.TissueAveraged.TPM.tsv"
```

2. Importing and manipulating the data
```{r}
library(MASS)
library(ggplot2)
library(ggrepel)
library(ComplexHeatmap)
library(circlize)
library(ggplot2)
library(grid)
library(gridExtra)

library(reshape2)
##Tissue Wide Expression Plots for GTEX
data<- read.delim(paste0(Prefix,"part2/",input1))#"ER.GTEX.TissueAveraged.TPM.tsv"

```

4. HEATMAP
```{r}
data <- join(data, EM_specificities, by = "Symbol")
data <- data[!is.na(data$Category),]
data <- data[!duplicated(data$Symbol),]
rownames(data)<- data[,1] #assign gene names as rownames  
data2<- data[,-c(1:3,11,35:37)] #remove the first three columns for the heatmap
data3<- log(data2+1)#Take log of the file (with a pseudocount)

data3[which(rowSums(data3)<0.01),]<- runif(ncol(data3), 0, 1)

data4 <- t(scale(t(data3)))#Normalize by row (by gene)
```


```{r}
pdf(paste0(Prefix,"part2/","human_results/heatmap.gtex.logzscaled_EM_substrate.pdf"),height=10,width=8)
Heatmap(data4, name = "z-scale log(TPM)", 
	#col = colorRamp2(c(-3,0,4), c("cadetblue3","floralwhite", "maroon4"),space = "RGB"), 
    #cluster_rows = TRUE, 
    col = colorRamp2(c(-3,-1.5,0,1.5,3), c("#2c7bb6","#abd9e9","floralwhite","#fdae61", "#d7191c"),space = "RGB"),
    cluster_columns = TRUE,
    column_title = "mRNA Expression of Human Tissues (GTEX)", 
    column_title_gp = gpar(fontsize = 10, fontface = "bold"),
    column_names_gp = gpar(fontsize = 7, fontface = "bold"),
    row_title = "ERs", row_title_rot = 90,
    row_title_gp = gpar(fontsize = 8, fontface = "bold"),
    cluster_rows = FALSE,
    show_row_names = TRUE,
    row_names_gp = gpar(fontsize = 5), #row names size
    #column_order = 1:dim(data4)[2],#Keep the column order, make clustering FALSE for this
    row_dend_side = "right", #Dendogram on the right side
    #row_order = 1:dim(data4)[1], #Keep the row order, make clustering FALSE for this
    show_column_dend = TRUE, #
    column_dend_side = "top",
    column_names_side = "bottom",
    split = data$Class, #Splitting by Class
    gap = unit(1, "mm"), #Gap
    )
dev.off()
```


5. PCA
```{r}
data_pca<-prcomp(data4,center=TRUE) #PCA 
data_out <- as.data.frame(data_pca$x) #X table of PCA
data_out$Category<- data$Category
data_out$Symbol<- rownames(data)
data_out$Category <- factor(data_out$Category , levels = unique(data_out$Category))#keep the order of species
##Calculation for percentage of variance explained by each component
eigs <- data_pca$sdev^2#Calculate percentage for PC values
percentage<- round(eigs/sum(eigs)*100,2)#Calculate percentage for PC values
percentage <- paste( colnames(data_out), "(", paste( as.character(percentage), "%", ")", sep="") ) #Calculate percentage for PC values
```


```{r}
## PLOT FOR ALL LABELS
pdf(paste0(Prefix,"part2/","human_results/pca_gtex.ER.substrate.labels.pdf"),height=5,width=7)
print(ggplot(data_out,aes(x=PC1,y=PC2,color=Category,label=Symbol))+
	scale_color_manual(values = c("darkgreen","grey","darkred","darkblue","pink","green1","red","steelblue1","purple","tan"))+
	geom_hline(yintercept = 0, lty = 2) +
	geom_vline(xintercept = 0, lty = 2)+
	geom_point(alpha = 0.8, size = 1.2)+
	geom_text(aes(label=Symbol),hjust=0, vjust=-1)+
	theme(panel.background = element_blank(),
		panel.border=element_rect(fill=NA),
		panel.grid.major = element_blank(),
		panel.grid.minor = element_blank(),
		strip.background=element_blank(),
		axis.text.x=element_text(colour="black"),
		axis.text.y=element_text(colour="black"),
		axis.ticks=element_line(colour="black"),
		plot.margin=unit(c(1,1,1,1),"line"))+
	xlab(percentage[1]) + ylab(percentage[2]) #Labels containing percentages
	)
dev.off()
```

```{r}
##PLOT FOR LOADINGS ALL LABELS
data_out_r <- as.data.frame(data_pca$rotation) #rotation data (loadings)
data_out_r$Symbol <- row.names(data_out_r) 
pdf(file=paste0(Prefix,"part2/","human_results/pca_gtex.loadings.ER.substrate.labels.pdf"),height=5,width=5)
print(ggplot(data_out_r,aes(x=PC1,y=PC2,label=Symbol))+
	geom_point()+
	geom_hline(yintercept = 0, lty = 2) +
	geom_vline(xintercept = 0, lty = 2)+
	geom_point(alpha = 0.8, size = 2)+
	geom_text(aes(label=Symbol),hjust=0, vjust=-1)+
	theme(panel.background = element_blank(),
		panel.border=element_rect(fill=NA),
		panel.grid.major = element_blank(),
		panel.grid.minor = element_blank(),
		strip.background=element_blank(),
		axis.text.x=element_text(colour="black"),
		axis.text.y=element_text(colour="black"),
		axis.ticks=element_line(colour="black"),
		plot.margin=unit(c(1,1,1,1),"line"))+
	xlab(percentage[1]) + ylab(percentage[2])
	)
dev.off()


```