---
title: "Domain"
output: html_notebook
---
```{r}
installed_pkgs <- installed.packages()

pkgs <-  c("GenVisR")
if (length(setdiff(pkgs, installed_pkgs)) > 0) {
    if (!require("BiocManager", quietly = TRUE))
    		install.packages("BiocManager")
    BiocManager::install(setdiff(pkgs, installed_pkgs))
}
pkgs <-  c("ggpubr")
if (length(setdiff("ggpubr", installed_pkgs)) > 0) {
    install.packages(pkgs = setdiff("igraph", installed_pkgs))
}
library(data.table)
library(ggpubr)
library(GenVisR)

Prefix <- "D:/OneDrive - ucas.ac.cn/论文/epi_reg/part7/"

set.seed(1)
```

```{r}
survival_dat <- read.table(paste0(Prefix,"TCGA_clinical.tsv"), header=T, sep="\t",quote = "")
colnames(survival_dat) <- c("times", "sample", "status", "cancer_type")
```


```{r}
library(data.table)
TCGA_mut <- fread("D:/GACEVI/ex_TCGA/TCGA_mutation_annotation.tsv", header = TRUE, sep="\t", select = c("var_class", "gene_name", "protein_change", "annotation_trans", "cell_line", "context"), quote = "")
```


```{r}
dmns_gnm_df <- read.table(paste0(Prefix, "genomics_pos_epidomains.txt"), header=TRUE, sep="\t", quote = "")
prts_df <- read.table(paste0(Prefix, "protein_info_epidomains.txt"), header=TRUE, sep="\t", quote = "")

TCGA_mut <- TCGA_mut[TCGA_mut$gene_name %in% DEG]
TCGA_mut <- merge(TCGA_mut, survival_dat, by.x = "cell_line", by.y= "sample")
TCGA_mut <- TCGA_mut[!(TCGA_mut$gene_name %in% c("TP53", "RB1")),]
TCGA_mut <- TCGA_mut[,1:8]
colnames(TCGA_mut) <- c("sample", "variant_class", "gene", "protein_change", "transcript", "cancer_type", "times", "status")

TCGA_mut_LAML <- TCGA_mut[TCGA_mut$cancer_type %in% c("LAML"),]
TCGA_mut_LAML <- TCGA_mut_LAML[TCGA_mut_LAML$protein_change != "",]
TCGA_mut_LAML <- as.data.frame(TCGA_mut_LAML)

TCGA_mut_BLCA <- TCGA_mut[TCGA_mut$cancer_type %in% c("BLCA","HNSC","SARC","CESC","BRCA"),]
TCGA_mut_BLCA <- TCGA_mut_BLCA[TCGA_mut_BLCA$protein_change != "",]
TCGA_mut_BLCA <- as.data.frame(TCGA_mut_BLCA)
write.table(TCGA_mut_BLCA, file=paste0(Prefix,"up_samples.txt"),sep="\t",
            col.names = TRUE,row.names = FALSE,quote=FALSE)
write.table(TCGA_mut_LAML, file=paste0(Prefix,"down_samples.txt"),sep="\t",
            col.names = TRUE,row.names = FALSE,quote=FALSE)
# waterfall(TCGA_mut_BLCA, fileType = "Custom", variant_class_order = c("Missense", "Nonsense", "Splice_Site", "3'UTR", "5'UTR", "Silent", "Intron"), maxGenes = 10)

```



```{r warning=FALSE}
Prefix <- "D:/OneDrive - ucas.ac.cn/论文/epi_reg/part7/"
TCGA_mut_BLCA <- read.table(paste0(Prefix, "up_samples.txt"), header=TRUE, sep="\t", quote = "")


```



```{r warning=FALSE}
data <- TCGA_mut_BLCA[TCGA_mut_BLCA$gene == "MAZ", c("gene", "protein_change", "transcript", "status")]

colnames(data) <- c("gene", "amino_acid_change", "transcript_name", "status")
data <- data[data$amino_acid_change != "",]

# Call lolliplot
up1 <- lolliplot(data, labelCol = "amino_acid_change", fillCol = "status")
```



```{r warning=FALSE}
data <- TCGA_mut_BLCA[TCGA_mut_BLCA$gene == "CBX2", c("gene", "protein_change", "transcript", "status")]

colnames(data) <- c("gene", "amino_acid_change", "transcript_name", "status")
data <- data[data$amino_acid_change != "",]

# Call lolliplot
up2 <- lolliplot(data, labelCol = "amino_acid_change", fillCol = "status")
```


```{r warning=FALSE}

data <- TCGA_mut_BLCA[TCGA_mut_BLCA$gene == "JADE1", c("gene", "protein_change", "transcript", "status")]

colnames(data) <- c("gene", "amino_acid_change", "transcript_name", "status")
data <- data[data$amino_acid_change != "",]

# Call lolliplot
down1 <- lolliplot(data, labelCol = "amino_acid_change", fillCol = "status")

```

```{r warning=FALSE}
data <- TCGA_mut_BLCA[TCGA_mut_BLCA$gene == "KAT2B", c("gene", "protein_change", "transcript", "status")]

colnames(data) <- c("gene", "amino_acid_change", "transcript_name", "status")
data <- data[data$amino_acid_change != "",]

# Call lolliplot
down2 <- lolliplot(data, labelCol = "amino_acid_change", fillCol = "status")

```



```{r}
pdf(file=paste0(Prefix,"lolliplot.pdf"),height=20,width=15)
ggarrange(up1, up2, down1, down2, ncol = 1, nrow = 4)
dev.off()
```